pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/leap/LeapDriver.java

```diff
@@ -95,33 +93,41 @@ public String initialState() {
         bucket.asScheduler().consume(1);
         final List<PersistentQueueElement> events = new ArrayList<>();
         final HttpClient client = HttpClient.newHttpClient();
-        final String uri = apiUri + "/ClientApi/V1/DeviceReadings";
-        final HttpRequest request = HttpRequest.newBuilder()
-            .uri(URI.create(uri))
-            .header("Authorization", "Bearer " + getAuthToken())
-            .build();
+        final String uri;
+        if (state == "")
+            uri = apiUri + "/ClientApi/V1/DeviceReadings";
+        else {
+            state = dateFormat.format(Date.from(dateFormat.parse(state).toInstant().plus(Duration.ofMillis(1))));
```

Comment:
Add a comment explaining why you are adding 1 ms. This will help future readers.

Reply:
Done